SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+    // Collect the indexess that has been selected.
+    for (int i = 0; i < SizeY; ++i) {
+      for (int j = 0; j < SizeX; ++j) {
+        selected_indexes.push_back(OffsetY * Width + i * Width * StrideY +
+                                   j * StrideX + OffsetX);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto selected_indexses_ptr = selected_indexes.begin();
+
+    for (int i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been selected and changed.
+      if (i < *selected_indexses_ptr) {
+        const DataT &retrieved = result[i];
+        const DataT &expected = initial_ref_data[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+      } else {
+        const DataT &retrieved = result[i];
+        const DataT expected = ref_data_for_change.front();
+        ref_data_for_change.erase(ref_data_for_change.begin());
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+        selected_indexses_ptr++;
+      }
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type) {
+    log::fail(TestDescriptionT(data_type), "Unexpected value at index ", i,
+              ", retrieved: ", retrieved, ", expected: ", expected,
+              ", with size x: ", SizeX, ", stride x: ", StrideX,
+              ", size y: ", SizeY, ", stride y: ", StrideY,
+              ", height: ", Height, ", offset x: ", OffsetX,
+              ", offset y: ", OffsetY, ", width: ", Width);
+
+    return false;
+  }
+};
+
+// Aliases to provide size or stride values to test.
+// This is the syntax sugar just for code readability.
+template <int N> using stride_type = std::integral_constant<int, N>;
+template <int N> using size_type = std::integral_constant<int, N>;
+template <int N> using offset_type = std::integral_constant<int, N>;
+template <int N> using simd_split_coeff = std::integral_constant<int, N>;
+
+template <typename TestCaseT, int SizeX, int SizeY, int StrideX, int StrideY,
+          int OffsetX, int OffsetY, int SimdSplitCoeff, typename... ArgsT>
+bool run_with_size_stride_offset(ArgsT &&...args) {
+  bool passed = for_all_combinations<
+      run_test, select_2d, size_type<SizeX>, size_type<SizeY>,
+      stride_type<StrideX>, stride_type<StrideY>, offset_type<OffsetX>,
+      offset_type<OffsetY>, simd_split_coeff<SimdSplitCoeff>>(
+      std::forward<ArgsT>(args)...);
+
+  return passed;
+}
+
+// Helping function that lets launch test with provided tested_types.
+// IMPORTANT: do not set offset value that greater than expected number of
+// strings or element in one string in simd_view.
```

Comment:
Is it possible to have `static_assert` for this condition somewhere?

Reply:
yes, they has been added in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/e81fee68457c2825f716260a4bdf088924b053c9